users/views.py

```diff
@@ -85,6 +85,37 @@ def _get_allowed_login_methods_for_client_id(client_id):
     return allowed_methods
 
 
+def _generate_final_login_methods(login_methods, next_url, idp_hint):
```

Comment:
Could something like `_generate_login_url_parameter_to_login_methods` be a better name for this function?

Reply:
The function does also some filtering, possibly dropping some login methods away. That's why I came up with the peculiar name.